app/views/courses/index.html.haml

```diff
@@ -1,8 +1,16 @@
 - provide(:title, 'courses')
 = link_to 'Main page', root_path
+%br/
+- if !user_signed_in?
+  = link_to 'Sign in', new_user_session_path
+  if you want to create your own course
 
 - @courses.each do |course|
-  %h3= link_to course.title, course_path(course)
+  %h3
+    - if course.belongs_to?(current_user) && user_signed_in?
```

Comment:
First check for user_signed_in. In this case there will not be situation when current_user will be nil.